sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/AzureSdkCompat/AzureEventSourceLogForwarder.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Diagnostics.Tracing;
+using Azure.Core.Diagnostics;
+using Azure.Core.Shared;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Azure;
+
+namespace Azure.Monitor.OpenTelemetry.AspNetCore.Internals.AzureSdkCompat
+{
+    internal sealed class AzureEventSourceLogForwarder : IHostedService, IDisposable
+    {
+        internal static readonly AzureEventSourceLogForwarder Noop = new AzureEventSourceLogForwarder(null);
+        private readonly ILoggerFactory _loggerFactory;
+
+        private readonly ConcurrentDictionary<string, ILogger> _loggers = new ConcurrentDictionary<string, ILogger>();
+
+        private readonly Func<EventSourceEvent, Exception, string> _formatMessage = FormatMessage;
+
+        private AzureEventSourceListener _listener;
+
+        public AzureEventSourceLogForwarder(ILoggerFactory loggerFactory)
+        {
+            _loggerFactory = loggerFactory;
+        }
+
+        private void LogEvent(EventWrittenEventArgs eventData)
+        {
+            var logger = _loggers.GetOrAdd(eventData.EventSource.Name, name => _loggerFactory!.CreateLogger(ToLoggerName(name)));
```

Comment:
If someone ends up getting the `Noop` instance and if this method gets invoked, there might be a `NullReferenceException`. Is this scenario even possible?

Reply:
if `LoggerFactory` is null, then listener that invokes `LogEvent` is never created. I also have a test that validates this case.